ISHPermissionKit/ISHPermissionsViewController.m

```diff
@@ -173,6 +160,47 @@ - (void)layoutChildViewControllerView:(UIView *)childView {
     [childView setCenter:CGPointMake(CGRectGetMidX(containerView.bounds), CGRectGetMidY(containerView.bounds))];
 }
 
+#pragma mark Permission state queries
+
+- (NSArray<ISHPermissionRequest *> *)permissionsRequestsForCategories:(NSArray<NSNumber *> *)categories passingTest:(BOOL (^_Nonnull)(ISHPermissionState))testBlock {
+    NSMutableArray<ISHPermissionRequest *> *requestsPassingTest = [NSMutableArray array];
+    BOOL dataSourceConfiguresRequests = [self.dataSource respondsToSelector:@selector(permissionsViewController:didConfigureRequest:)];
+
+    for (NSNumber *boxedCategory in categories) {
+        ISHPermissionRequest *request = [ISHPermissionRequest requestForCategory:boxedCategory.unsignedIntegerValue];
+        
+        if (dataSourceConfiguresRequests && request.allowsConfiguration) {
+            [self.dataSource permissionsViewController:self didConfigureRequest:request];
+        }
+
+        if (testBlock([request permissionState])) {
```

Comment:
Should we extend the `permissionState` documentation to include a hint that for requests that allow configuration, this property must be handled with care?